### PR TITLE
Move travel pay controller logic to service

### PR DESCRIPTION
**Pull request by athifw**
## Summary

Moves travel pay controller logic to service, to make testing travel pay controllers easier.
Adds tests and removes unused files to hit 80% code coverage.

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/78913
https://github.com/department-of-veterans-affairs/va.gov-team/issues/81889

## Testing done

- New code is tested by existing controller tests.

Screenshot 2024-07-01 at 11 16 28 AM

### CODE REVIEW

File: modules/travel_pay/app/controllers/travel_pay/pings_controller.rb [hunk 1]
```diff
@@ -26,5 +20,9 @@ def authorized_ping
     def client
```
Comment (kjduensing) [inline modules/travel_pay/app/controllers/travel_pay/pings_controller.rb:20]: We don't need to create a client if we're relying on the service

File: modules/travel_pay/app/services/travel_pay/service.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module TravelPay
+  class Service
```
Comment (kjduensing) [inline modules/travel_pay/app/services/travel_pay/service.rb:4]: I think we're almost there. I'd like to have the client simply be a wrapper around Faraday that bundles requests together:

In the client:
```ruby
def get_claims(current_user)
  veis_token = self.request_veis_token
  sts_token = self.request_sts_token(current_user)
  btsss_token = self.request_btsss_token(veis_token, sts_token)

  # Faraday will return the response object or throw a Faraday::Error
  return request_claims(veis_token, btsss_token)
end
```

In the service:

```ruby
def get_claims(current_user)
  claims_response = client.get_claims(current_user)
  
  # do all the business logic-y stuff to the claims response (initial sort, parse, etc)

  return sorted_claims;
end
```

Comment (kjduensing) [reply modules/travel_pay/app/services/travel_pay/service.rb:4]: What do you think about this? What are the drawbacks?

**Comment by kjduensing** [comment]
I expected to see some changes to the tests. How did you ensure the logic works the same?

File: modules/travel_pay/app/services/travel_pay/service.rb (hunk 2 above)
Comment (athifw) [reply modules/travel_pay/app/services/travel_pay/service.rb:4]: I think this makes sense - with the benefit that adding specs for the service can also test the claims sort logic. The service doesn't do a whole lot right now, so I don't see the drawbacks.

**Comment by athifw** [comment]
> @athifw I expected to see some changes to the tests. How did you ensure the logic works the same?

Currently, all the tests still work since all the controller tests rely on VCR, so as long as the requests are the same, the tests will still pass.

File: config/settings/test.yml [hunk 3]
```diff
@@ -434,7 +434,7 @@ travel_pay:
     resource: 'resource_id'
     tenant_id: 'tenant_id'
     auth_url: 'https://auth.veis.gov'
-  mock: true
+  mock: false
```
Comment (athifw) [inline config/settings/test.yml:437]: to test the sts assertion service. all tests should be mocked or use vcr cassettes anyway

File: modules/travel_pay/app/services/travel_pay/client.rb [hunk 4]
```diff
@@ -173,6 +148,41 @@ def veis_params
...
+
+    def request_claims(veis_token, btsss_token)
+      btsss_url = Settings.travel_pay.base_url
+      api_key = Settings.travel_pay.subscription_key
+
+      connection(server_url: btsss_url).get('api/v1/claims') do |req|
+        req.headers['Authorization'] = "Bearer #{veis_token}"
+        req.headers['BTSSS-Access-Token'] = btsss_token
+        req.headers['Ocp-Apim-Subscription-Key'] = api_key
+        req.headers['X-Correlation-ID'] = SecureRandom.uuid
```
Comment (kjduensing) [inline modules/travel_pay/app/services/travel_pay/client.rb:182]: Not for this PR unless you're feeling motivated, but at some point we should pull this header stuff into a separate method.

**Review by kjduensing — APPROVED** [review]
Looks good, thanks @athifw 🌵

File: modules/travel_pay/app/services/travel_pay/client.rb (hunk 4 above)
Comment (athifw) [reply modules/travel_pay/app/services/travel_pay/client.rb:182]: I'm ok with doing this at a later time!

Comment (LindseySaari) [reply modules/travel_pay/app/services/travel_pay/client.rb:182]: It seems like the connection method could be pulled into a [configuration class](https://github.com/department-of-veterans-affairs/vets-api) as we see in many other of the service class patterns in Vets API unless theres some reason not to use this config. The service/config class pattern seems much cleaner

Comment (kjduensing) [reply modules/travel_pay/app/services/travel_pay/client.rb:182]: I've had some problems with the Config class in the past, stemming from the fact that the Configuration base class is a singleton. We're choosing a simpler (more repeat-y) approach for now until we can figure out if the existing config pattern is going to work for us. Let us know if this blocks an approval and we can figure out a game plan.

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**